l10n_es_aeat_mod349/models/account_invoice.py

```diff
@@ -0,0 +1,165 @@
+        # Invoices by period
+        elif period_selection in ['1T', '2T', '3T', '4T']:
+            if not period_id:
+                raise exceptions.Warning(
+                    _('Cannot get invoices.\nThere is no period selected'))
+            domain.append(('period_id', 'in', period_id))
+        # Invoices by month
+        else:
+            if not month and not fiscalyear:
+                raise exceptions.Warning(
+                    _('Cannot get invoices.\nThere is no month and/or fiscal '
+                      'year selected'))
+            month = int(month)
+            year = self._get_year_from_fy_month(fiscalyear, month)
+            month_last_day = calendar.monthrange(year, month)[1]
+            date_start = datetime(year=year, month=month, day=1)
+            date_stop = datetime(year=year, month=month, day=month_last_day)
+            domain.append(
+                ('date_invoice', '>=', fields.Date.to_string(date_start)))
+            domain.append(
+                ('date_invoice', '<=', fields.Date.to_string(date_stop)))
+        return self.search(domain)
+
+    @api.multi
+    def clean_refund_invoices(
+            self, partner, fiscalyear=None, periods=None, month=None,
+            period_selection=None):
+        """Separate refunds from invoices"""
+        invoices = self.env['account.invoice']
+        refunds = self.env['account.invoice']
+        for inv in self:
+            if inv.type in ('in_refund', 'out_refund'):
+                if not inv.origin_invoices_ids:
+                    invoices += inv
+                    continue
+                for origin_line in inv.origin_invoices_ids:
+                    if (origin_line.state in ('open', 'paid') and
+                            origin_line.partner_id.commercial_partner_id ==
+                            partner):
+                        if period_selection == '0A':
+                            if (origin_line.period_id.id not in
+                                    [period.id for period in
+                                     fiscalyear.period_ids if not
+                                     period.special]):
+                                refunds += inv
+                            else:
+                                invoices += inv
+                        elif period_selection in ['1T', '2T', '3T', '4T']:
+                            if origin_line.period_id not in periods:
+                                refunds += inv
+                            else:
+                                invoices += inv
+                        else:
+                            month = int(month)
+                            year = self._get_year_from_fy_month(fiscalyear,
+                                                                month)
+                            if (fields.Date.from_string(
+                                    origin_line.date_invoice) <
+                                    datetime(year=year, month=month, day=1)):
```

Comment:
TypeError: can't compare datetime.datetime to datetime.date
Para arreglarlo basta con añadir la importación de date y sustituir por
 ```python
date(year=year, month=month, day=1)):
```